File: backend/internal/model/globalstats.go
package model

type GlobalStats struct {
	Nations []string         `json:"nations"`
	Types   []string         `json:"types"`
	Cells   []GlobalStatCell `json:"cells"`
}

type GlobalStatCell struct {
	Nation      string  `json:"nation"`
	Type        string  `json:"type"`
	Count       int     `json:"count"`
	AvgBR       float64 `json:"avg_br"`
	WinRate     float64 `json:"win_rate"`
	GamesPlayed int     `json:"games_played"`
	PlayerCount int     `json:"player_count"`
}

File: frontend/src/api/globalstats.ts
import { request } from './index'

export interface GlobalStatCell {
  nation: string
  type: string
  count: number
  avg_br: number
  win_rate: number
  games_played: number
  player_count: number
}

export interface GlobalStats {
  nations: string[]
  types: string[]
  cells: GlobalStatCell[]
}

export function getGlobalStats() {
  return request<GlobalStats>('/globalstats')
}
